### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'split')

### DISCUSSION THREAD

**Discussion by jasonhilldm** [vikejs/vike#1906] [Discussion] [answered]
I am having a weird problem with a test app when running a production build and deploying to a node web app hosted in Azure. I am getting the following error:

```
TypeError: Cannot read properties of undefined (reading 'split')
     at assertReact (/home/site/wwwroot/node_modules/react-streaming/dist/cjs/server/renderToStream.js:136:52)
     at Object.<anonymous> (/home/site/wwwroot/node_modules/react-streaming/dist/cjs/server/renderToStream.js:41:1)
     at Module._compile (node:internal/modules/cjs/loader:1358:14)
     at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
     at Module.load (node:internal/modules/cjs/loader:1208:32)
     at Module._load (node:internal/modules/cjs/loader:1024:12)
     at Module.require (node:internal/modules/cjs/loader:1233:19)
     at require (node:internal/modules/helpers:179:18)
     at Object.<anonymous> (/home/site/wwwroot/node_modules/react-streaming/dist/cjs/server/index.js:8:26)
     at Module._compile (node:internal/modules/cjs/loader:1358:14)

The error could be a CJS/ESM issue, see https://vike.dev/broken-npm-package
```

I don't get this issue when running the app locally in dev. I have also tried running the build locally and then deploying the artefacts to a locally hosted docker container and it works fine. It's just when I deploy the site to the node instance in Azure that I get the above error. I am using Circle CI to perform the Azure deployment and this runs npm install as one of the deployment steps so are there any platform variances that could be causing this?

I have tried some of the steps outlined on the page https://vike.dev/broken-npm-package but I wasn't able to resolve the issue. Any idea what could be causing this?

**Comment by brillout** [comment]
That's quite unexpected indeed. Are you using's Preact's React compat layer? Can you publish a reproduction? Alternatively, dig into why `version` is undefined at `/home/site/wwwroot/node_modules/react-streaming/dist/cjs/server/renderToStream.js:136:52`. Note that `server_1` comes from React, so it should define `version`.

**Reply by jasonhilldm** [reply]
I moved away from Preact entirely and went back to regular React. I had previously deployed another Vike app to my Azure web app and it worked fine, so maybe the combination of packages in this new test app is causing some issues.

I did notice that Auth.js has a dependency on preact and preact-render-to-string and was wondering if they would cause any problems. I know I had issues with externals when trying to use Preact with Vike and had to make some modules noExternal in the ssr config. I don't have any ssr settings in this app though as I am just using React.

I just tried deploying a local production build from my Mac to the Azure web app and it worked, so it seems to be an issue during the Circle CI build & deployment process.

For clarification, I have a copy of my package.json in dist/server and run npm install in there during Circle CI deployment. This means I have a fully constructed node-modules folder that gets deployed to the Azure web app. Is that OK?

Anyway, I will try and dig into that version issue that you flagged and can also try and create a repro.

**Reply by jasonhilldm** [reply, marked as answer]
I used a different container in Circle CI for the deployment and now it all works great. Not sure what the issue was though!
